### ISSUE TITLE
How to handle duplicate primary keys?

### ISSUE THREAD

**Issue by xbroak** [realm/realm-java#1446]
Is there an exception type i can catch when a duplicate value tries to get inserted? I don't want to have to check manually before every write. I might aswell not have the primary key do anything at all if that were the case!

**Comment by kneth** [comment]
If the primary key already exists, a `RealmException` will be thrown.

**Comment by cmelchior** [comment]
See also https://github.com/realm/realm-java/issues/1416

For now you can do something like

```
} catch (RealmException e) {
  if (e.getMessage().startsWith("Primary key constraint broken") {
    // Handle it...
  }
}

**Comment by kneth** [comment]
I will close the issue as @cmelchior has provided a work-around.

**Comment by xbroak** [comment]
Surely a workaround is not a solution? As such it should be left open? :)

**Comment by cmelchior** [comment]
Hi @xbroak. We do intend to fix it, but it is being tracked in #1416 instead.
